includes/embeds/class-amp-core-block-handler.php

```diff
@@ -224,14 +226,36 @@ public function ampify_video_block( $block_content, $block ) {
 	 * @return string Filtered block content.
 	 */
 	public function ampify_cover_block( $block_content, $block ) {
-		if ( isset( $block['attrs']['backgroundType'] ) && 'video' === $block['attrs']['backgroundType'] ) {
-			$block_content = preg_replace(
-				'/(?<=<video\s)/',
-				'layout="fill" object-fit="cover" ',
-				$block_content
-			);
-		}
-		return $block_content;
+		return preg_replace_callback(
+			'#<(?:img|video)(?= )[^>]*? class="[^"]*?wp-block-cover__(?:image|video)-background[^"]*?"#',
```

Comment:
This regex would match class names like `my-custom-class-wp-block-cover__image-background` or `wp-block-cover__video-background-plugin-baz`, which is unintentional. Here's one way to resolve the issue:

```suggestion
			'#<(?:img|video)(?= )[^>]*? class=(?:"|"[^"]*? )wp-block-cover__(?:image|video)-background(?:"| [^"]*?")#',
```

Reply:
This seems to be a bit better:

```suggestion
			'#<(?:img|video)(?= )[^>]*? class="(?:[^"]*? )?wp-block-cover__(?:image|video)-background(?: [^"]*?)?"#',
```

Reply:
That works, and takes less steps to match too!